jsaddle/src/Language/Javascript/JSaddle/Run.hs

```diff
@@ -181,9 +198,25 @@ runJavaScript sendReqAsync = do
           case mThisSync of
             Nothing -> putStrLn $ "Rsp_Sync: " <> show syncReqId <> " not found"
             Just thisSync -> putMVar thisSync ()
+      sendReqAsync req = do
+        count <- atomically $ do
+          modifyTVar' pendingReqs ((:) req)
```

Comment:
This puts the new request before the other pending requests. Is that intentional? It may be more straightforward to just do `(++) [req]` and remove the `reverse reqs` below.

Reply:
Yes its intentional, this TVar will be updated frequently by many threads, so the modifyTVar' should happen quickly.